src/main/java/ch/njol/skript/util/PersistentDataUtils.java

```diff
@@ -0,0 +1,593 @@
+
+			// This is to store the data on the ItemType or TileState
+			if (holder instanceof ItemType) {
+				((ItemType) holder).setItemMeta((ItemMeta) actualHolder);
+			} else if (actualHolder instanceof TileState) {
+				((TileState) actualHolder).update();
+			}
+		} else if (holder instanceof Metadatable) { // Set as Metadata instead
+			((Metadatable) holder).setMetadata(name, new FixedMetadataValue(Skript.getInstance(), value));
+		}
+	}
+
+	/**
+	 * Sets the Persistent Data Tag's value for the given holder from the given name and value.
+	 * This method will check the holder's {@linkplain PersistentDataContainer} and Metadata.
+	 * @param holder The holder of the Persistent Data Tag. See {@linkplain PersistentDataUtils#getActualHolder(Object)}
+	 * @param name The name of the single variable (e.g. <b>"myVariable" from {myVariable}</b>)
+	 * @see PersistentDataUtils#getSingle(Object, String)
+	 * @see PersistentDataUtils#setSingle(Object, String, Object)
+	 */
+	public static void removeSingle(Object holder, String name) {
+		if (name.contains(Variable.SEPARATOR)) // This is a list variable..
+			return;
+
+		PersistentDataHolder actualHolder = getActualHolder(holder);
+		if (actualHolder == null)
+			return;
+
+		NamespacedKey key = getNamespacedKey(name);
+		if (key == null)
+			return;
+
+		if (actualHolder.getPersistentDataContainer().has(key, SINGLE_VARIABLE_TYPE)) { // Can be serialized, try to remove Persistent Data
+			actualHolder.getPersistentDataContainer().remove(key);
+
+			// This is to store the data on the ItemType or TileState
+			if (holder instanceof ItemType) {
+				((ItemType) holder).setItemMeta((ItemMeta) actualHolder);
+			} else if (actualHolder instanceof TileState) {
+				((TileState) actualHolder).update();
+			}
+		} else if (holder instanceof Metadatable) { // Try to remove Metadata instead
+			((Metadatable) holder).removeMetadata(name, Skript.getInstance());
+		}
+	}
+
+	/*
+	 * List Variable Modification Methods
+	 */
+
+	/**
+	 * Gets the Persistent Data Tag's value of the given list variable name from the given holder.
+	 * This method may return a single value, or multiple, depending on the given name.
+	 * If the value set was not serializable, it was set under Metadata and is retrieved from Metadata here.
+	 * @param holder The holder of the Persistent Data Tag. See {@linkplain PersistentDataUtils#getActualHolder(Object)}
+	 * @param name The name of the list variable (e.g. <b>"myList::*" from {myList::*}</b>)
+	 * @return The Persistent Data Tag's value(s) from the holder, or an empty array if: 
+	 * the holder was invalid, the name was invalid, the key was invalid, or if no value(s) could be found.
+	 * @see PersistentDataUtils#setList(Object, String, Object)
+	 * @see PersistentDataUtils#removeList(Object, String)
+	 * @see PersistentDataUtils#getListMap(Object, String)
+	 */
+	@SuppressWarnings({"null", "unchecked"})
+	public static Object[] getList(Object holder, String name) {
```

Comment:
why not consolidate getListMap and getList into one? you can always replicate this method with `getListMap().values()`

Reply:
well - not exactly

getListMap doesn't acceopt specific indexes, while getList does
however, there is some some code repetition which could be removed, so I'll make some changes 😄